File: python_solutions/boj_2828.py
import sys
input = sys.stdin.readline

n, m = map(int, input().split())
j = int(input())
apple = [int(input()) for _ in range(j)]
start, end = 1, m

ans = 0
for pos in apple:
    if start <= pos <= end:
        continue
    elif pos < start:
        dist = start-pos
        ans += dist
        start -= dist
        end -= dist
    else:
        dist = pos-end
        ans += dist
        start += dist
        end += dist

print(ans)

File: cpp_solutions/boj_2828/boj_2828.cpp
#include <iostream>

using namespace std;

int main(){
    int N, M, J; cin >> N >> M >> J;

    int l = 1, r = M, dist = 0;

    while (J--){
        int apple, tmp = 0; cin >> apple;

        if (apple < l) {
            tmp = l - apple;
            l -= tmp; r -= tmp;
        } else if (apple > r) {
            tmp = apple - r;
            l += tmp; r += tmp;
        }
        dist += tmp;
    }

    cout << dist;
}